src/helper/component/BaseTab.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <div class="tab">
+    <button
+      v-for="{ value, color, icon } in items"
+      :key="value"
+      :class="getClass('tab-item', value)"
+      :style="{ backgroundColor: color }"
+      @click="click(value)"
+    >
+      <!-- icon -->
+      <img
+        v-if="isSelected(value)"
+        :src="require(`../../assets/icons/${icon}.svg`)"
+        class="icon"
+      />
+
+      <!-- text -->
+      <span class="text">{{ value }}</span>
+
+      <!-- ribbon -->
+      <div :class="getClass('line', value)" />
+    </button>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'BaseTab',
+
+  props: {
+    items: { type: Array, required: true },
+    value: { type: String, default: '' },
+  },
+
+  methods: {
+    isSelected(value) {
+      return this.value === value
```

Comment:
name of two variables are 'value'. I think you can choose better names for them!

Reply:
because both variables are value of tab